setup.py

```diff
@@ -774,15 +772,26 @@ def write_version(filename: str = os.path.join(*[my_dir, "airflow", "git_version
     'zendesk': ["zendesk"],
 }
 
-
-# Make devel_all contain all providers + extras + unique
-devel_all = list(
+# All "users" extras (no devel extras)
+all_ = list(
     set(
-        devel
-        + [req for req_list in EXTRAS_REQUIREMENTS.values() for req in req_list]
+        [req for req_list in EXTRAS_REQUIREMENTS.values() for req in req_list]
         + [req for req_list in PROVIDERS_REQUIREMENTS.values() for req in req_list]
     )
 )
+EXTRAS_REQUIREMENTS.update({"all": all_})
+
+# Development extras
+EXTRAS_REQUIREMENTS.update(
+    {
+        'devel': devel_minreq,  # includes doc
+        'devel_hadoop': devel_hadoop,  # includes devel_minreq
+        'doc': doc,
+    }
+)
```

Comment:
```suggestion
EXTRAS_REQUIREMENTS.update(
    {
        'all': all_
        'devel': devel_minreq,  # includes doc
        'devel_hadoop': devel_hadoop,  # includes devel_minreq
        'doc': doc,
    }
)
```

(It just looks a bit odd having it split out in two statements. Either way, having update for a single key is particularly odd)

Reply:
Oh, indeed it's weird 👀